### ISSUE TITLE
Expose a `url` HttpRequest.Builder mutation 

### ISSUE THREAD

**Issue by kevincianfarini** [apollographql/apollo-kotlin#4037]
**Use case**
I'm configuring an interceptor that will dynamically choose server host based on some criteria. In OkHttp it's trivial to configure an interceptor which rewrites the host. In Apollo, this is impossible with `HttpRequest.Builder`. 

I've had to hack around this with the following function. 

```kt
private fun HttpRequest.newBuilderWithUrl(newUrl: String) = HttpRequest.Builder(
    url = newUrl,
    method = this.method
).apply {
    val body = body // name shadow required because public property in different module.
    if (body != null) body(body)
    addHeaders(headers)
}
```

**Describe the solution you'd like**
Expose a `HttpRequest.Builder.url(url: String)` function. 

Others have asked on this issue board about this feature, but presumably since they weren't using MPP Apollo they just circumvented the issue with an OkHttp interceptor. See #3579

**Comment by martinbonnin** [comment]
Ooohh good catch! I'll add this in a few.

**Comment by BoD** [comment]
Available in v3.3.0.
